src/util.jl

```diff
@@ -287,3 +287,19 @@ function switch_ring(v::MPolyElem, ring::MPolyRing)
 end
 
 # ------------------------------------------------------------------------------
+function check_eq_coeffs(eq)
+    q = Deque{Dict}()
+    push!(q, ModelingToolkit.Symbolics.value(eq.rhs.dict))
```

Comment:
it won't always have a dict? Better to act on the `arguments(eq.rhs)`

Reply:
Got it, changed it to work with arguments.